runtime/test/planner-tests.js

```diff
@@ -698,6 +699,37 @@ describe('SearchTokensToParticles', function() {
   });
 });
 
+describe('MatchRecipeByVerb', function() {
+  it('removes a particle and adds a recipe', async () => {
+    let manifest = await Manifest.parse(`
+      recipe
+        particle can jump
+      
+      schema Feet
+      schema Energy
+
+      particle JumpingBoots in 'A.js'
+        JumpingBoots(in Feet f, in Energy e)
+      particle FootFactory in 'B.js'
+        FootFactory(out Feet f)
+      particle NuclearReactor in 'C.js'
+        NuclearReactor(out Energy e)
+
+      recipe jump
+        JumpingBoots.f <- FootFactory.f
+        JumpingBoots.e <- NuclearReactor.e         
+    `);
+
+    let arc = createTestArc('test-plan-arc', manifest, 'dom');
+    let strategizer = {generated: [{result: manifest.recipes[0], score: 1}]};
+    let mrv = new MatchRecipeByVerb(arc);
+    let {results} = await mrv.generate(strategizer);
+    assert.equal(results.length, 1);
+    assert.equal(results[0].result.particles.length, 0);
+    assert.deepEqual(results[0].result.toString(), 'recipe\n  JumpingBoots.e -> NuclearReactor.e\n  JumpingBoots.f -> FootFactory.f');
```

Comment:
that serialization looks weird, I can't imagine we would serialize it like that. does this test really run to completion?

Reply:
ohh, they are unresolved recipe constraints

Reply:
Yes. I copied the serialization that we output. You need an imagination upgrade ;)